src/cogs/snipe.py

```diff
@@ -25,11 +27,20 @@ def __init__(self, bot):
         self._message_history: dict[int, list[discord.Message]] = defaultdict(list)
 
     @commands.hybrid_command(with_app_command=True)
-    async def snipe(self, ctx: Context, number: int = 0):
-        if ctx.channel.id not in self._deleted:
+    async def snipe(self, ctx: Context, number: int = 0, channel: Snipeable | None = None):
+        req_perms = discord.Permissions(send_messages = True, view_channel = True)
+
+        if channel and (
+            not isinstance(ctx.author, discord.Member)
+            or not req_perms.is_subset(channel.permissions_for(ctx.author))
```

Comment:
I don't like this. Regular users should not be able to snipe cross-channel. I have concerns about this because it could be abused for raids. If at all I'd allow this only for staff so we can snipe to an internal channel, but I don't really see the point of that either.

Clearing other channels is nice though.

Reply:
I don't really see how you could abuse it for raids more than snipe already tbh.